### PR TITLE
Jgit dependency through git client

### PR DESCRIPTION
**Pull request by gounthar**
Please don't merge this PR, this is just a proof of concept, a base to discuss.

After creating #48, I experimented with a potential solution, though I'm not certain if it aligns with Maven best practices.

Here's what I tried:

1. I believe I managed to fetch the `jgit` dependency with the same version as the `git-client` plugin by referencing its `pom.xml`.
   (The project compiles successfully, which suggests it's working)

2. To keep this approach current, we would need to maintain the version number:
   ```xml
   <git.client.version>5.0.0</git.client.version>``
   
Tools like Dependabot or Renovate could help us keep this version up to date.
What are your thoughts on this approach? I'm particularly interested in:
- Whether this aligns with Maven best practices
- Any potential issues or benefits you foresee
- Alternative approaches we might consider

I look forward to hearing your feedback and suggestions.


### Testing done

`mvn clean install`

### Submitter checklist
- [X] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [X] Ensure that the pull request title represents the desired changelog entry
- [X] Please describe what you did
- [X] Link to relevant issues in GitHub or Jira
- ~~Link to relevant pull requests, esp. upstream and downstream changes~~
- [X] Ensure you have provided tests - that demonstrates feature works or fixes the issue

<!--
Put an `x` into the [ ] to show you have filled the information.
The template comes from https://github.com/jenkinsci/.github/blob/master/.github/pull_request_template.md 
You can override it by creating .github/pull_request_template.md in your own repository 
-->

### CODE REVIEW

**Comment by sridamul** [comment]
Hey Bruno, I had tested the changes locally and seems to work fine. IMO Initially, we can stick to this and experiment other approaches like "utilizing updatecli" later.

File: pom.xml [hunk 1]
```diff
@@ -165,6 +166,49 @@
             </plugins>
         </pluginManagement>
         <plugins>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-dependency-plugin</artifactId>
+                <version>3.7.1</version>
```
Comment (jonesbusy) [inline pom.xml:172]: If we go with this solution, I would move the plugin version on the `</pluginManagement>` and execute the plugin only on the core pom

File: pom.xml [hunk 2]
```diff
@@ -165,6 +166,49 @@
...
+                                </artifactItem>
+                            </artifactItems>
+                        </configuration>
+                    </execution>
+                </executions>
+            </plugin>
+            <plugin>
+                <groupId>org.codehaus.mojo</groupId>
+                <artifactId>properties-maven-plugin</artifactId>
+                <version>1.2.1</version>
```
Comment (jonesbusy) [inline pom.xml:197]: If we go with this solution, I would move the plugin version on the `</pluginManagement>` and execute the plugin only on the core pom

**Comment by sridamul** [comment]
Hey Bruno, I have inspected a few past PRs from Dependabot to update Jgit dependency in the Git-client-plugin. From my observation, I found that there are only regressions either related to the Jenkins core or the plugin itself. If I am not wrong, then using Dependabot in our tool will not cause any issue (Since, neither core nor plugin involved in our tool). wdyt? (Please correct me if I am wrong) Thanks!

**Comment by gounthar** [comment]
I appreciate your perspective. Upon reflection, I realize that my previous suggestion was based on a gut feeling rather than concrete data.

Given this, I agree that we should proceed with the following approach:

1. Start by implementing Dependabot for our version management.
2. Monitor the results over the next few months.
3. If we encounter any issues, we can reassess our strategy and consider alternatives.

This approach allows us to:
- Begin with a widely-used, automated solution
- Gather real-world data on its effectiveness for our specific use case
- Remain flexible and open to adjustments if needed

Does this sound like a reasonable plan to everyone? I'm open to any thoughts or concerns you might have.

**Comment by jonesbusy** [comment]
Sure fine for me

**Pull request closed without merging**